databuilder/extractor/feast_extractor.py

```diff
@@ -0,0 +1,139 @@
+     * columns are features stored in the feature table
+    """
+
+    FEAST_SERVICE_CONFIG_KEY = "instance_name"
+    FEAST_ENDPOINT_CONFIG_KEY = "endpoint"
+    DESCRIBE_FEATURE_TABLES = "describe_feature_tables"
+    DEFAULT_CONFIG = ConfigFactory.from_dict(
+        {FEAST_SERVICE_CONFIG_KEY: "main", DESCRIBE_FEATURE_TABLES: True}
+    )
+
+    def init(self, conf: ConfigTree) -> None:
+        conf = conf.with_fallback(FeastExtractor.DEFAULT_CONFIG)
+        self._feast_service = conf.get_string(FeastExtractor.FEAST_SERVICE_CONFIG_KEY)
+        self._describe_feature_tables = conf.get_bool(
+            FeastExtractor.DESCRIBE_FEATURE_TABLES
+        )
+        self._client = Client(
+            core_url=conf.get_string(FeastExtractor.FEAST_ENDPOINT_CONFIG_KEY)
+        )
+        self._extract_iter: Union[None, Iterator] = None
+
+    def get_scope(self) -> str:
+        return "extractor.feast"
+
+    def extract(self) -> Union[TableMetadata, None]:
+        """
+        For every feature table from Feast, a multiple objets are extracted:
+
+        1. TableMetadata with feature table description
+        2. Programmatic Description of the feature table, containing
+           metadata - date of creation and labels
+        3. Programmatic Description with Batch Source specification
+        4. (if applicable) Programmatic Description with Stream Source
+           specification
+        """
+        if not self._extract_iter:
+            self._extract_iter = self._get_extract_iter()
+        try:
+            return next(self._extract_iter)
+        except StopIteration:
+            return None
+
+    def _get_extract_iter(self) -> Iterator[TableMetadata]:
+        for project in self._client.list_projects():
+            for feature_table in self._client.list_feature_tables(project=project):
+                yield from self._extract_feature_table(project, feature_table)
+
+    def _extract_feature_table(
+        self, project: str, feature_table: FeatureTable
+    ) -> Iterator[TableMetadata]:
+        columns = []
+        for index, entity_name in enumerate(feature_table.entities):
+            entity = self._client.get_entity(entity_name, project=project)
+            columns.append(
+                ColumnMetadata(
+                    entity.name, entity.description, entity.value_type, index
+                )
+            )
+
+        for index, feature in enumerate(feature_table.features):
+            columns.append(
+                ColumnMetadata(
+                    feature.name,
+                    None,
+                    feature.dtype.name,
+                    len(feature_table.entities) + index,
+                )
+            )
+
+        yield TableMetadata(
+            "feast",
+            self._feast_service,
+            project,
+            feature_table.name,
+            None,
+            columns,
+        )
+
+        if self._describe_feature_tables:
```

Comment:
interesting to know you yield the prog description in the same extractor.

Reply:
Actually it was quite easy, since both tables and prog descriptions use same class. I was thinking once about creating multiple extractors - one for table and second for prog descirptions, but decided to not overcomplicate.

Reply:
cc @dikshathakur3119 for FYI as this will be easier for Lyft internal programmatic description use case.

Reply:
it is super useful